### ISSUE TITLE
Enhancement in `Meshsync api` to include `kind` filter param

### ISSUE THREAD

**Issue by ShivamCoder23** [meshery/meshery#10455]
### Current Behavior

Like connection api has param for status and kind to filter([doc](https://docs.meshery.io/reference/rest-apis#get-handle-get-request-for-getting-all-connections)) same enhancement might be needed in meshsync so the filtering the data based on kind will be more consistent . Because currently we are directly applying filter which backend doesn't support so page and pagesize doesn't come in effect.

### Mockups

---
### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

**Comment by leecalcote** [comment]
Please enhance the description of this issue in following with the provided template. Offer links to APIs that are being referenced. You can use the API reference in Meshery Docs or links in meshery/meshery source.

**Comment by anmolchhabra21** [comment]
Let me check for this and get an idea via API docs.

**Comment by leecalcote** [comment]
@sudhanshutech  are you going to take care of this?

**Comment by adityaofficial10** [comment]
@leecalcote  Can I work on this?

**Comment by paranik-jpg** [comment]
Hi @sudhanshutech @leecalcote, this enhancement appears to be already implemented. `GetMeshSyncResources` in `server/handlers/meshsync_handler.go` parses `kind := r.URL.Query()["kind"]` and applies it via `filterByKinds(query, kind)` (added in commit 1798a51b0f1). 

Could a maintainer please verify and close this issue?

**Comment by leecalcote** [comment]
Correct, @paranik-jpg. Thanks for confirming. You know what we need now?

1. An update in Meshery Docs.
2. A blog post on the topic.

Both are open opportunities for you to undertake. Execute in order, if you're interested.

**Comment by paranik-jpg** [comment]
Hey @leecalcote, makes sense. I can take on the docs update first. I'll check out meshery.io and put together a PR covering the kind filter. Where in the docs would you prefer this to sit, or should I just open an issue over there first to track it?

Can look into the blog post once this is sorted.
